core/relation/status.go

```diff
@@ -0,0 +1,12 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package relation
+
+// Status describes the status of a relation.
+type Status string
+
+const (
+	Active  Status = "active"
+	Revoked Status = "revoked"
+)
```

Comment:
Status is work in progress - followup PRs will add extra doc etc. The idea is that while a relation is Alive, it can has different status values. Alive and Active means that everything is ok. Alive and "Revoked" means that the relation has not been removed/killed, but is not able to function at the moment. The exact terminology is yet to be decided. It may be "Inactive" depending on user feedback.
Status is needed because an entity's life can only go from alive->dying->dead. Once a relation's life is not alive anymore, status becomes irrelevant. This is similar to status as used for machines and units.